1st-Mile: How Tos

undavide opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

- Repurpose existing "Develop > Use Cases" into "Develop -> HowTos"
- Rationalize the HowTos IA
- Add more content.

ToDo: as soon as the Playground implements Script Mode, repurpose the snippets to be copy-paste-able in it.
## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

hollyschinsky commented on src/pages/guides/develop/how_tos/page_metadata.md:
````diff
@@ -0,0 +1,89 @@
+});
+```
+
+### Page range Example
````
Make range caps too?

hollyschinsky commented on src/pages/guides/develop/how_tos/page_metadata.md:
```diff
@@ -0,0 +1,89 @@
+
+The returned value is always an array of [`PageMetadata`](../../../references/addonsdk/app-document.md#pagemetadata) objects.
+
+### Single page Example
```
Make page caps too, or example not caps? (not sure the rules here :/)

hollyschinsky reviewed (changes requested):
Just a few things I noticed we need to update...

nimithajalal commented on src/pages/guides/develop/how_tos/element_metadata.md:
```diff
@@ -0,0 +1,95 @@
+  - https://github.com/undavide
+  - https://github.com/hollyschinsky
+---
+# Element Metadata
```
Similar comment as document metadata

undavide replied:
I have changed the parent "Using Metadata" -> "Use Metadata", but I won't add "Use" to the children, because it's ugly/redundant:

```
How-to
└─ Use Metadata
   ├─ Use Document Metadata ❌
   ├─ Use Page Metadata     ❌
   └─ Use Element Metadata  ❌

How-to
└─ Use Metadata
   ├─ Document Metadata ✅
   ├─ Page Metadata     ✅
   └─ Element Metadata  ✅
```

I would even say that "Metadata" in the children is ugly/redundant too, so we may even go with:

```
How-to
└─ Use Metadata
   ├─ Document 
   ├─ Page     
   └─ Element  
```

But I won't fight over it :-)

nimithajalal replied:
Second option is all gtg

nimithajalal commented on src/pages/guides/develop/how_tos/grouping_elements.md:
```diff
@@ -0,0 +1,240 @@
+  - https://github.com/hollyschinsky
+---
+
+# Grouping Elements
```
Similar comment as document metadata

nimithajalal commented on src/pages/guides/develop/how_tos/imports_and_constants.md:
```diff
@@ -0,0 +1,19 @@
+  - https://github.com/undavide
+  - https://github.com/hollyschinsky
+---
+# Imports and Constants
```
What is the objective of this? Can we rename the heading to make it a directive to achieve a specific end?

nimithajalal commented on src/pages/guides/develop/how_tos/listening_to_events.md:
```diff
@@ -0,0 +1,19 @@
+  - https://github.com/undavide
+  - https://github.com/hollyschinsky
+---
+# Listening to Events
```
Listen to Events

nimithajalal commented on src/pages/guides/develop/how_tos/local_data_management.md:
```diff
@@ -0,0 +1,60 @@
+  - https://github.com/undavide
+  - https://github.com/hollyschinsky
+---
+# Storing Data
```
Store Data

nimithajalal commented on src/pages/guides/develop/how_tos/modal_dialogs.md:
```diff
@@ -0,0 +1,170 @@
+  - https://github.com/hollyschinsky
+---
+
+# Modal Dialogs
```
What is the objective of this? Can we rename the heading to make it a directive to achieve a specific end? 

Example:

Use Modal Dialogs?

nimithajalal replied:
What is the objective of this? Can we rename the heading to make it a directive to achieve a specific end? 

Example:

Use Modal Dialogs?

nimithajalal commented on src/pages/guides/develop/how_tos/monetization_flows.md:
```diff
@@ -0,0 +1,47 @@
+  - https://github.com/undavide
+---
+
+# Monetization Flows
```
What is the objective of this? Can we rename the heading to make it a directive to achieve a specific end? 

Use monetization flows in `something`

undavide replied:
I've deleted the file, it was an unlinked copy and the content is already available elsewhere

nimithajalal commented on src/pages/guides/develop/how_tos/oauth2.md:
```diff
@@ -0,0 +1,136 @@
+  - https://github.com/undavide
+---
+
+# Using OAuth 2.0
```
Use oAuth2.0

nimithajalal commented on src/pages/guides/develop/how_tos/page_metadata.md:
```diff
@@ -0,0 +1,89 @@
+  - https://github.com/undavide
+  - https://github.com/hollyschinsky
+---
+# Page Metadata
```
What is the objective of this? Can we rename the heading to make it a directive to achieve a specific end? 

I think the second heading is the objective. You can take a call, though.

undavide replied:
See [this comment](https://github.com/AdobeDocs/express-add-ons-docs/pull/104#discussion_r1983358570)

nimithajalal commented on src/pages/guides/develop/how_tos/positioning_elements.md:
```diff
@@ -0,0 +1,179 @@
+  - https://github.com/undavide
+---
+
+# Positioning Elements
```
What is the objective of this? Can we rename the heading to make it a directive to achieve a specific end? 

Example 

Position elements `somewhere`

undavide replied:
I have changed to "Position elements" according to the naming convention, but won't add "where", as it's not really necessary 
- how to position elements > wherever users want/need 
- how to position elements "on the canvas" (other choices are not really possible)

nimithajalal commented on src/pages/guides/develop/how_tos/premium_content.md:
```diff
@@ -0,0 +1,194 @@
+
+import '/src/styles.css'
+
+# Managing Premium Content
```
Manage `your?` Premium Content

nimithajalal commented on src/pages/guides/develop/how_tos/replacing_elements.md:
```diff
@@ -0,0 +1,19 @@
+  - https://github.com/undavide
+  - https://github.com/hollyschinsky
+---
+# Replacing Elements
```
What is the objective of this? Can we rename the heading to make it a directive to achieve a specific end? 

Example: 

Replace elements `somewhere`?

undavide replied:
See [this comment](https://github.com/AdobeDocs/express-add-ons-docs/pull/104#discussion_r1983406410)

nimithajalal commented on src/pages/guides/develop/how_tos/ui_document_api_communication.md:
```diff
@@ -0,0 +1,19 @@
+  - https://github.com/undavide
+  - https://github.com/hollyschinsky
+---
+# Add-on UI and Document API communication
```
What is the objective of this? Can we rename the heading to make it a directive to achieve a specific end?

nimithajalal commented on src/pages/guides/develop/how_tos/user_info.md:
```diff
@@ -0,0 +1,47 @@
+  - https://github.com/undavide
+  - https://github.com/hollyschinsky
+---
+# Identifying Users
```
Identify users

Can we add more context in the heading? What does the reader achieve after implementing this how-to?

nimithajalal commented on src/pages/guides/develop/how_tos/using_audio.md:
```diff
@@ -0,0 +1,71 @@
+  - https://github.com/hollyschinsky
+---
+
+# Using Audio
```
Use Audio `somewhere?`

undavide replied:
See [this comment](https://github.com/AdobeDocs/express-add-ons-docs/pull/104#discussion_r1983406410)

nimithajalal commented on src/pages/guides/develop/how_tos/using_color.md:
```diff
@@ -0,0 +1,123 @@
+  - https://github.com/hollyschinsky
+---
+
+# Using Color
```
Same comment as above

nimithajalal commented on src/pages/guides/develop/how_tos/using_geometry.md:
```diff
@@ -0,0 +1,227 @@
+  - https://github.com/undavide
+---
+
+# Using Geometry
```
Same comment as above

nimithajalal commented on src/pages/guides/develop/how_tos/using_images.md:
```diff
@@ -0,0 +1,105 @@
+  - https://github.com/hollyschinsky
+---
+
+# Using Images
```
Same comment as above

nimithajalal commented on src/pages/guides/develop/how_tos/using_text.md:
```diff
@@ -0,0 +1,458 @@
+  - https://github.com/hollyschinsky
+---
+
+# Using Text
```
Same comment as above

nimithajalal commented on src/pages/guides/develop/how_tos/using_videos.md:
```diff
@@ -0,0 +1,66 @@
+  - https://github.com/hollyschinsky
+---
+
+# Using Videos
```
Use Videos `somewhere?`

undavide replied:
See [this comment](https://github.com/AdobeDocs/express-add-ons-docs/pull/104#discussion_r1983406410)

nimithajalal commented on src/pages/guides/develop/how_tos.md:
Consider the how-to as a recipe. 
A recipe has a clear, defined end. It addresses a specific question. It shows someone - who can be assumed to have some basic knowledge already - how to achieve something.

Can we add that the developer knows certain items and lists them?

undavide replied:
I would agree to switch to the base form of the verb instead of the gerund, but I will not add "How to" to every title. It is horribly redundant 😁 and a waste of precious side-bar space.
Besides, all main page headings should match the navbar; hence, it would result in  the following:
```
How-to
├─ How to author Adobe Express content
├─ How to create renditions
├─ How to use images
├─ How to...
```
This is unnecessary and—given how poorly our navbar deals with strings—leads to ugly carriage returns and double lines.
Deal? :-)

nimithajalal replied:
Oh my bad. I meant to say, including the folder name, the url would how-to- 'do something'

undavide commented on src/pages/guides/develop/how_tos/document_metadata.md:
```diff
@@ -0,0 +1,54 @@
+  - https://github.com/undavide
+  - https://github.com/hollyschinsky
+---
+# Document Metadata
```
See comment on Element Metadata

undavide commented:
@nimithajalal I should have addressed all your comments to the best of my abilities. 
I believe we'll have a How-tos round #2 when the Playground implements Script mode (I'll have to add a tab to each code snippets for a copy/paste-able, Playground-ready version). It's likely that new how-tos will be added, if devs give feedback (maybe DevChamps?)
@hollyschinsky let's wait for Nimitha's green light before merging, we're still targeting Friday 7th. 🤞🏼
